pkg/config/sysprep_test.go

```diff
@@ -83,35 +84,36 @@ var _ = Describe("SysprepConfigMap", func() {
 		},
 	})
 
-	Describe("With invalid file name", func() {
-		BeforeEach(func() {
-			createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", "wrongname.xml"))
-		})
+	DescribeTable("Assert CreateSysprepDisks behaviour",
+		func(filenames []string, shouldSucceed bool) {
 
-		It("Should fail on creating config map iso disk", func() {
-			err := CreateSysprepDisks(vmiConfigMap, false)
-			Expect(err).To(HaveOccurred())
-		})
-	})
+			for _, f := range filenames {
+				createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", f))
+			}
 
-	Describe("With valid file name (autounattend.xml)", func() {
-		BeforeEach(func() {
-			// Check case-insensitivity (should accept anything Windows accepts).
-			createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", "AutounattenD.xml"))
-		})
-
-		It("Should create a new config map iso disk", func() {
-			err := CreateSysprepDisks(vmiConfigMap, false)
-			Expect(err).NotTo(HaveOccurred())
-			_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
-			Expect(err).NotTo(HaveOccurred())
-		})
-
-		It("Should create a new secret iso disk", func() {
-			err := CreateSysprepDisks(vmiSecret, false)
-			Expect(err).NotTo(HaveOccurred())
-			_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
-			Expect(err).NotTo(HaveOccurred())
-		})
-	})
+			if shouldSucceed {
+				err := CreateSysprepDisks(vmiConfigMap, false)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
+				Expect(err).NotTo(HaveOccurred())
+
+				err = CreateSysprepDisks(vmiSecret, false)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
```

Comment:
Apologies, this is now outside of an It() so the iso from the first run isn't removed for us, I'll rewrite this on Monday.